instrumentation/opentelemetry-instrumentation-openai/src/opentelemetry/instrumentation/openai/patch.py

```diff
@@ -0,0 +1,308 @@
+            )
+            set_span_attribute(
+                self.span,
+                "gen_ai.usage.total_tokens",
+                self.prompt_tokens + self.completion_tokens,
+            )
+            set_event_completion(
+                self.span,
+                [
+                    {
+                        "role": "assistant",
+                        "content": "".join(self.result_content),
+                    }
+                ],
+            )
+
+            self.span.set_status(StatusCode.OK)
+            self.span.end()
+            self._span_started = False
+
+    def __enter__(self):
+        self.setup()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.cleanup()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            chunk = next(self.stream)
+            self.process_chunk(chunk)
+            return chunk
+        except StopIteration:
+            self.cleanup()
+            raise
+
+    def set_response_model(self, chunk):
+        if self.response_model:
+            return
+
+        if getattr(chunk, "model", None):
+            self.response_model = chunk.model
+
+    def set_response_id(self, chunk):
+        if self.response_id:
+            return
+
+        if getattr(chunk, "id", None):
+            self.response_id = chunk.id
+
+    def build_streaming_response(self, chunk):
+        if getattr(chunk, "choices", None) is None:
+            return
+
+        choices = chunk.choices
+        content = []
+        if not self.function_call and not self.tool_calls:
+            for choice in choices:
+                if choice.delta and choice.delta.content is not None:
+                    content = [choice.delta.content]
+
+        elif self.function_call:
+            for choice in choices:
+                if (
+                    choice.delta
+                    and choice.delta.function_call is not None
+                    and choice.delta.function_call.arguments is not None
+                ):
+                    content = [choice.delta.function_call.arguments]
+
+        elif self.tool_calls:
+            for choice in choices:
+                if choice.delta and choice.delta.tool_calls is not None:
+                    toolcalls = choice.delta.tool_calls
+                    content = []
+                    for tool_call in toolcalls:
+                        if (
+                            tool_call
+                            and tool_call.function is not None
+                            and tool_call.function.arguments is not None
+                        ):
+                            content.append(tool_call.function.arguments)
+
+        for choice in choices:
+            finish_reason = choice.finish_reason
+            if finish_reason:
+                set_span_attribute(
+                    self.span,
+                    GenAIAttributes.GEN_AI_RESPONSE_FINISH_REASONS,
```

Comment:
same as  in the other comment, this attribute value should be an array of strings, not the first finish_reason